os/net/routing/rpl-classic/rpl-timers.c

```diff
@@ -180,23 +181,23 @@ handle_dio_timer(void *ptr)
 #endif /* RPL_CONF_STATS */
       dio_output(instance, NULL);
     } else {
-      PRINTF("RPL: Suppressing DIO transmission (%d >= %d)\n",
+      LOG_DBG("Suppressing DIO transmission (%d >= %d)\n",
              instance->dio_counter, instance->dio_redundancy);
     }
     instance->dio_send = 0;
-    PRINTF("RPL: Scheduling DIO timer %lu ticks in future (sent)\n",
+    LOG_DBG("Scheduling DIO timer %lu ticks in future (sent)\n",
            instance->dio_next_delay);
     ctimer_set(&instance->dio_timer, instance->dio_next_delay, handle_dio_timer, instance);
   } else {
     /* check if we need to double interval */
     if(instance->dio_intcurrent < instance->dio_intmin + instance->dio_intdoubl) {
       instance->dio_intcurrent++;
-      PRINTF("RPL: DIO Timer interval doubled %d\n", instance->dio_intcurrent);
+      LOG_DBG("DIO Timer interval doubled %d\n", instance->dio_intcurrent);
     }
     new_dio_interval(instance);
   }
 
-#if DEBUG
+#if LOG_DBG_ENABLED
```

Comment:
rpl-lite uses `if (LOG_DBG_ENABLED) {/* .. */ }`. I prefer the use of preprocessor as it discards the entire body and makes it easier for optimizer to eliminate dead code. That being said, I'm not sure exactly what the optimizer does when encountering a `if` with a constant condition.